spec/requests/v1/api_spec.rb

```diff
@@ -367,6 +367,256 @@
+          expect(task.task_items.count).to eq(1)
+        end
+      end
+
+      context "when the token is not valid" do
+        before { @response = post "/api/v1/tasks/#{task.id}/task_items", params, empty_header }
+
+        it "should return a HTTP 401 status" do
+          expect(@response.status).to eq(401)
+        end
+
+        it "should create nothing" do
+          expect(task.task_items.count).to eq(0)
+        end
+      end
+    end
+
+    describe "GET /api/v1/tasks/:id/task_items" do
+      let(:task) do
+        Task.create! do |t|
+          t.name = 'New Task'
+          t.url = 'http://www.gsa.gov'
+          t.user_id = user.id
+          t.app_id = client_app.id
+          t.task_items = [
+            TaskItem.new(name: "Task Item #1", "external_id": "abcdef"),
+            TaskItem.new(name: "Task Item #2", url: 'http://valid_url.com')
+          ]
+        end
+      end
+
+      context "when the token is valid" do
+        before { @response = get "/api/v1/tasks/#{task.id}/task_items", nil, header }
+
+        it "should return a HTTP 200 status" do
+          expect(@response.status).to eq(200)
+        end
+
+        it "should return all the task_items associated with that task in an array" do
+          expect(@response.status).to eq(200)
+          parsed_json = JSON.parse(@response.body)
+          expect(parsed_json.length).to eq(2)
+          expect(parsed_json.first["name"]).to eq("Task Item #1")
+          expect(parsed_json.last["name"]).to eq("Task Item #2")
+        end
+      end
+
+      context "when the token is invalid" do
+        before { @response = get "/api/v1/tasks/#{task.id}/task_items", nil, empty_header }
+
+        it "should return a 401 HTTP status" do
+          expect(@response.status).to eq(401)
+        end
+      end
+    end
+
+    describe "PUT /api/v1/tasks/:id/task_items/:item_id" do
+      let(:task) do
+        Task.create! do |t|
+          t.name = 'New Task'
+          t.url = 'http://www.gsa.gov'
+          t.user_id = user.id
+          t.app_id = client_app.id
+          t.task_items = [
+            TaskItem.new(name: "Task Item #1", "external_id": "abcdef"),
+            TaskItem.new(name: "Task Item #2", url: 'http://valid_url.com')
+          ]
+        end
+      end
+
+      let(:task_item) { task.task_items.first }
+      let(:params) { {task_item: { name: "Task Item Changed", "complete": true} } }
+
+      context "when the token is valid" do
+        before { @response = put "/api/v1/tasks/#{task.id}/task_items/#{task_item.id}", params, header }
+
+        it "should return a HTTP 200 status code" do
+          expect(@response.status).to eq(200)
+        end
+
+        it "should return JSON representing the new task item" do
+          parsed_json = JSON.parse(@response.body)
+          expect(parsed_json["name"]).to eq(params[:task_item][:name])
+          expect(parsed_json["external_id"]).to eq(task_item.external_id)
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.